### PR TITLE
Support loading worlds from CLI

### PR DESCRIPTION
**Pull request by mugling**
Automatically load first save in a world via `./cataclysm --world foo`. Mostly intended to aid developers via more efficient testing. Windows developers can probably define a shortcut target to their preferred testing save.

In a separate PR it might be worth collaborating on a standard prototyping world that could also double for usage in the unit tests.

Future PR will probably also add support for quickly toggling `--mods` for use when testing mods.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Sounds useful - most of the time I keep loading the same world over and over.

Which world name is used? Path name or true world name?

**Comment by mugling** [comment]
> Sounds useful - most of the time I keep loading the same world over and over.

Exactly the same for me. We should probably gradually move towards a standard test world

> Which world name is used? Path name or true world name?

True world name

**Comment by codemime** [comment]
The idea occurred to me too. It will prove useful.

File: src/game.cpp [hunk 1]
```diff
@@ -3615,6 +3615,28 @@ void game::load_uistate(std::string worldname)
     read_from_file_optional( savefile, uistate );
 }
 
+bool game::load( const std::string &world ) {
+    auto opts = world_generator->get_all_worlds();
+    if( opts.find( world ) == opts.end() ) {
+        debugmsg( "unknown world '%s'", world.c_str() );
+        return false;
+    }
+    auto saves = opts[world]->world_saves;
```
Comment (codemime) [inline src/game.cpp]: You can find an iterator only once and don't repeat the lookup.

**Comment by mugling** [comment]
Ready

**Comment by Coolthulhu** [comment]
Seems that it doesn't skip the initial data load, meaning it goes through JSONs twice.
This could be skipped.

**Pull request merged**